### PR TITLE
Review suggestions for V2 rulesets

### PR DESCRIPTION
**Pull request by evanweible-wf**

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/v2.recommended.yaml [hunk 1]
```diff
@@ -84,17 +83,14 @@ analyzer:
     avoid_returning_null_for_void: warning
     avoid_setters_without_getters: warning
     avoid_slow_async_io: warning
+    avoid_type_to_string: warning
```
Comment (dustyholmes-wf) [inline lib/v2.recommended.yaml:86]: I hadn't seen this one, but I think it makes sense. 

the .toString in these situations is just extra information to parse.

File: lib/v2.recommended.yaml [hunk 2]
```diff
@@ -84,17 +83,14 @@ analyzer:
...
     avoid_types_on_closure_parameters: warning
     avoid_unused_constructor_parameters: warning
     cascade_invocations: warning
     comment_references: warning
     control_flow_in_finally: warning
     curly_braces_in_flow_control_structures: warning
     directives_ordering: warning
-    duplicate_ignore: warning
-    invalid_required_named_param: warning
-    invalid_required_positional_param: warning
```
Comment (dustyholmes-wf) [inline lib/v2.recommended.yaml:96]: Why did you remove these here? I presumed you would just remove them from the lints section.

File: lib/v2.recommended.yaml [hunk 3]
```diff
@@ -135,15 +131,15 @@ analyzer:
     unawaited_futures: warning
     unnecessary_await_in_return: warning
     unnecessary_brace_in_string_interps: warning
-    unnecessary_cast: warning
```
Comment (dustyholmes-wf) [inline lib/v2.recommended.yaml:138]: Same on this one.

File: lib/v2.recommended.yaml (hunk 2 above)
Comment (evanweible-wf) [reply lib/v2.recommended.yaml:96]: I moved them into `v2.yaml`

File: lib/v2.recommended.yaml (hunk 3 above)
Comment (evanweible-wf) [reply lib/v2.recommended.yaml:138]: I moved them into `v2.yaml`

File: lib/v2.recommended.yaml (hunk 2 above)
Comment (dustyholmes-wf) [reply lib/v2.recommended.yaml:96]: Ah, these got lifted to required.

File: lib/v2.recommended.yaml (hunk 1 above)
Comment (evanweible-wf) [reply lib/v2.recommended.yaml:86]: More than that, using `runtimeType` as a String doesn't work as expected in minified code, so any check relying on that behavior likely won't work. For debugging purposes, we should still be able to just log the object itself rather than its type.

File: lib/v2.recommended.yaml (hunk 2 above)
Comment (paysonlippert-wk) [reply lib/v2.recommended.yaml:96]: What criteria do you use for what is in the base version and what is in recommended? I agree on keeping them, just curious why the move?

File: lib/v2.recommended.yaml [hunk 4]
```diff
@@ -135,15 +131,15 @@ analyzer:
...
     unnecessary_getters_setters: warning
     unnecessary_lambdas: warning
     unnecessary_parenthesis: warning
     unnecessary_overrides: warning
+    unnecessary_string_escapes: warning
+    unnecessary_string_interpolations: warning
```
Comment (paysonlippert-wk) [inline lib/v2.recommended.yaml:139]: 👍

**Review by dustyholmes-wf — APPROVED** [review]

**Review by paysonlippert-wk — APPROVED** [review]
I think these changes make sense, I am unclear on what belongs in the base options and what is recommended, but I trust your reasoning

File: lib/v2.recommended.yaml (hunk 2 above)
Comment (evanweible-wf) [reply lib/v2.recommended.yaml:96]: I've always tried to use this: https://github.com/Workiva/workiva_analysis_options/#how-lints-and-analyzer-rules-are-evaluated

But it is still admittedly subjective. Given these are all very easy to fix and (IMO) shouldn't be very controversial, I thought required was reasonable, but I'm happy to discuss keeping any of them in recommended

Comment (paysonlippert-wk) [reply lib/v2.recommended.yaml:96]: That makes sense to me, I am happy to have them required.
